pyseir/rt/infer_rt.py

```diff
@@ -0,0 +1,647 @@
+
+rt_log = structlog.get_logger(__name__)
+
+
+def run_rt_for_fips(fips):
+    """Entry Point for Infer Rt"""
+
+    # Generate the Data Packet to Pass to RtInferenceEngine
+    input_df = _generate_input_data(fips, include_testing_correction=True)
+
+    # Save a reference to instantiated engine (eventually I want to pull out the figure generation
+    # and saving so that I don't have to pass a display_name and fips into the class
+    engine = RtInferenceEngine(
+        data=input_df, display_name=_get_display_name(fips), fips=fips, include_deaths=False
+    )
+
+    # Generate the output DataFrame (consider renaming the function infer_all to make it clearer)
+    output_df = engine.infer_all()
+
+    # Save the output to json for downstream repacking and incorporation.
+    county_output_file = get_run_artifact_path(fips, RunArtifact.RT_INFERENCE_RESULT)
+    if output_df is not None and not output_df.empty:
+        output_df.to_json(county_output_file)
+
+
+def _get_display_name(fips) -> str:
+    """Need to find the right function for this. Right now just return the fips"""
+    return str(fips)
+
+
+def _generate_input_data(fips, include_testing_correction=True):
+    """
+    Allow the RtInferenceEngine to be agnostic to aggregation level by handling the loading first
+
+    include_testing_correction: bool
+        If True, include a correction for testing increases and decreases.
+    """
+    times, observed_new_cases, observed_new_deaths = load_data.load_new_case_data_by_fips(
+        fips, t0=InferRtConstants.REF_DATE, include_testing_correction=include_testing_correction,
+    )
+
+    date = [InferRtConstants.REF_DATE + timedelta(days=int(t)) for t in times]
+    df = pd.DataFrame(dict(cases=observed_new_cases, deaths=observed_new_deaths), index=date)
+    return df
+
+
+class RtInferenceEngine:
+    """
+    This class extends the analysis of Bettencourt et al to include mortality data in a
+    pseudo-non-parametric inference of R_t.
+
+    Parameters
+    ----------
+    data: DataFrame
+        DataFrame with a Date index and at least one "cases" column.
+    include_deaths: bool
+        If True, include the deaths timeseries in the calculation. XCorrelated and Averaged
+    display_name: str
+        Needed for Figures. Should just return figures along with dataframe and then deal with title
+        and save location somewhere downstream.
+    fips: str
+        Just used for output paths. Should remove with display_name later.
+    """
+
+    def __init__(
+        self, data, display_name, fips, include_deaths=False,
+    ):
+
+        self.dates = data.index
+        self.cases = data.cases
+        self.deaths = data.deaths if "deaths" in data else None
+
+        self.include_deaths = include_deaths
+        self.display_name = display_name
+        self.fips = fips
+
+        # Load the InferRtConstants (TODO: turn into class constants)
+        self.r_list = InferRtConstants.R_BUCKETS
```

Comment:
I would prefer if you passed in these params as something like 

```
def __init__(self, config: InferRtConfig = InferRtConfig({}):
    self.config = config

def infer_all(self, fips, data, ...):
    
```

This will definitely require a bit more refactoring, but I think it gets to a much better separation of concerns.

Reply:
I would still prefer to inject these values rather than directly accessing the class here. Also, I think that it would be helpful to just have a variable `self.config = rt_config` or even `self.config = InferRtConstants`, to remove the need to write out all the variables twice in the init